packages/client/src/Components/Mapping/helpers/planningRouteFor.js

```diff
@@ -54,57 +62,99 @@ function bearingBetween (/* latLng */ p1, /* latLng */ p2) {
 /** we plot a line marker perpendicular to the direction
  * of travel at route waypoints.  Determine the marker to use for this bearing
  */
-function bearingMarkerFor (/* number */ angle) {
+function bearingMarkerFor(/* number */ angle) {
   let icon
-
-  if (angle === 180 || angle === 360 || angle === 0) {
-    // for the 180 and 360, which are flat icons so end icon is used
-    icon = turnEnd
-  } else if (angle > 0 && angle < 90) {
-    // North to East
-    icon = turnLeft
-  } else if (angle < 180) {
-    // East to South
-    icon = turnRight
-  } else if (angle < 260) {
-    // South to West
-    icon = turnLeft
-  } else if (angle < 271) {
-    // South
-    icon = turnEnd
-  } else if (angle < 360) {
-    // West to North
-    icon = turnRight
+  var iconAngles = {
+    turn0degVar: {icon: turn0deg, markerAngle: 0},
+    turn30degVar: {icon: turn30deg, markerAngle: 30},
+    turn60degVar: {icon: turn60deg, markerAngle: 60},
+    turn90degVar: {icon: turn90deg, markerAngle: 90},
+    turn120degVar: {icon: turn120deg, markerAngle: 120},
+    turn150degVar: {icon: turn150deg, markerAngle: 150},
+    turnLightVar: {icon: lightTurn, markerAngle: null},
+    turnBackVar: {icon: backTurn, markerAngle: null},
+  }
```

Comment:
actually @lm93547  the complex if structure can now be replaced by a `switch` statement, now that we're matching exact angles.